R/collectors.R

```diff
@@ -214,33 +214,41 @@ guess_parser <- function(x, locale = default_locale(), guess_integer = FALSE, na
 
 #' Parse factors
 #'
-#' `parse_factor` is similar to [factor()], but will generate
-#' warnings if elements of `x` are not found in `levels`.
+#' `parse_factor()` is similar to [factor()], but generates a warning if
+#' `levels` have been specified and some elements of `x` are not found in those
+#' `levels`.
 #'
-#' @param levels Character vector providing set of allowed levels. if `NULL`,
-#'   will generate levels based on the unique values of `x`, ordered by order
-#'   of appearance in `x`.
+#' @param levels Character vector of the allowed levels. When `levels = NULL`
+#'   (the default), `levels` are discovered from the unique values of `x`, in
+#'   the order in which they are encountered in `x`.
 #' @param ordered Is it an ordered factor?
-#' @param include_na If `NA` are present, include as an explicit factor to level?
+#' @param include_na If `TRUE` and `x` contains at least one `NA`, then `NA`
+#'   is included in the levels of the constructed factor.
```

Comment:
This is still not totally correct or complete, but I think it's an improvement. Once we sort out #1389, we can revisit this.